### ISSUE TITLE
preprocess crash

### ISSUE THREAD

**Issue by Kerushii** [axolotl-ai-cloud/axolotl#800]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

Preprocessing local dataset going as smoothly as the default mistral qlora config using a huggingface dataset

### Current behaviour

As soon as the dataset field in the mistral qlora config changes to datasets:
  - path: sample.json
    type: completion
    ds_type: json
preprocessing won't succeed
![Screenshot from 2023-10-28 22-31-30](https://github.com/OpenAccess-AI-Collective/axolotl/assets/85707358/cc7f5359-ac87-4770-b966-22a41c305d4f)

### Steps to reproduce

1. Make sure mistral fine tunes fine using the default qlora config
2. change the config datasets field to datasets:
  - path: sample.json
    type: completion
    ds_type: json
3. rerun and watch it burns

### Config yaml

This is really just default config with the above bug triggering change:
```
base_model: mistralai/Mistral-7B-v0.1
model_type: MistralForCausalLM
tokenizer_type: LlamaTokenizer
is_mistral_derived_model: true

load_in_8bit: false
load_in_4bit: true
strict: false

datasets:
  - path: sample.json
    type: completion
    ds_type: json
dataset_prepared_path: last_run_prepared
val_set_size: 0.01
output_dir: ./qlora-out

adapter: qlora
lora_model_dir:

sequence_len: 8192
sample_packing: true
pad_to_sequence_len: true

lora_r: 32
lora_alpha: 16
lora_dropout: 0.05
lora_target_linear: true
lora_fan_in_fan_out:
lora_target_modules:
  - gate_proj
  - down_proj
  - up_proj
  - q_proj
  - v_proj
  - k_proj
  - o_proj

wandb_project:
wandb_entity:
wandb_watch:
wandb_run_id:
wandb_log_model:
dataset_processes: 1
gradient_accumulation_steps: 4
micro_batch_size: 2
num_epochs: 1
optimizer: adamw_bnb_8bit
lr_scheduler: cosine
learning_rate: 0.0002

train_on_inputs: false
group_by_length: false
bf16: true
fp16: false
tf32: false

gradient_checkpointing: true
early_stopping_patience:
resume_from_checkpoint:
local_rank:
logging_steps: 1
xformers_attention:
flash_attention: true

warmup_steps: 10
eval_steps: 0.05
eval_table_size:
eval_table_max_new_tokens: 128
save_steps:
debug:
deepspeed:
weight_decay: 0.0
fsdp:
fsdp_config:
special_tokens:
  bos_token: "<s>"
  eos_token: "</s>"
  unk_token: "<unk>"

```

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10

### axolotl branch-commit

main/8b79ff0

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by Kerushii** [comment]
https://pastebin.com/9dn15wmT

the following dmesgs were generated whenever that happens

**Comment by winglian** [comment]
Can you run the preprocessing with python instead of accelerate?

**Comment by Kerushii** [comment]
> Can you run the preprocessing with python instead of accelerate?

    def process(self, dataset):
        features = dataset.features.keys()
        num_proc = min(64, 1)
        map_kwargs = {}
        if self.prompt_tokenizer.supports_batched:
            map_kwargs["batched"] = True
            map_kwargs["batch_size"] = 100
        return dataset.map(
            self.prompt_tokenizer.tokenize_prompt,
            num_proc=num_proc,
            remove_columns=features,
            **map_kwargs,
        )

this change is hacky, but solved the problem. This might be related to the fact that this is running on 13900kf.
I also noticed that dataset_process is not respected in config
Running using CUDA_VISIBLE_DEVICES=1 accelerate launch -m axolotl.cli.train  examples/mistral/qlora.yml --report_to tensorboard does not work; same error. It can only run with num process =1

**Comment by NanoCode012** [comment]
Hm, I thought we had a config to set the num_proc manually via yaml.

**Comment by NanoCode012** [comment]
Re-reading this, it seems that it's due to some OOM. We have a config `dataset_processes` to manually set this now.

https://github.com/OpenAccess-AI-Collective/axolotl/blob/89134f2143cd3325802813eb97cd05c783932201/src/axolotl/datasets.py#L47-L49
